extensions/amp-date-display/1.0/amp-date-display.js

```diff
@@ -86,14 +87,33 @@ AmpDateDisplay['passthrough'] = true;
 
 /** @override */
 AmpDateDisplay['props'] = {
+  'datetime': {
+    attrs: ['datetime', 'timestamp-ms', 'timestamp-seconds', 'offset-seconds'],
+    parseAttrs: parseDateAttrs,
+  },
   'displayIn': {attr: 'display-in'},
-  'offsetSeconds': {attr: 'offset-seconds', type: 'number'},
   'locale': {attr: 'locale'},
-  'datetime': {attr: 'datetime'},
-  'timestampMs': {attr: 'timestamp-ms', type: 'number'},
-  'timestampSeconds': {attr: 'timestamp-seconds', type: 'number'},
 };
 
+/**
+ * @param {!Element} element
+ * @return {?number}
+ * @visibleForTesting
+ */
+export function parseDateAttrs(element) {
+  const datetimeStr = element.getAttribute('datetime');
+  const datetime = parseDate(datetimeStr);
+  const timestampMs = Number(element.getAttribute('timestamp-ms'));
+  const timestampSeconds =
+    Number(element.getAttribute('timestamp-seconds')) * 1000;
+  const offsetSeconds = Number(element.getAttribute('offset-seconds')) * 1000;
```

Comment:
Here at least we should check for `NaN` or default to `0` since we are using it with a sum later. I don't think we should rely on the AMP validator to disallow non-integer attribute values since this isn't guaranteed to be used on a validated page.

Reply:
I added a check for `NaN`, but tbh not sure it's all that needed. `Number(null)` and `Number('')` both give `0`.  `Number('something very invalid')` yields `NaN` and maybe it'd have been better to just have the broken state than recover to `0`?

Reply:
Good point, I guess it is reasonable to allow it to break so it does not obscure the issue.